src/frontend/src/btc/services/btc-pending-sent-transactions.services.ts

```diff
@@ -0,0 +1,34 @@
+import { btcPendingSentTransactionsStore } from '$btc/stores/btc-pending-sent-transactions.store';
+import { getPendingBtcTransactions } from '$lib/api/backend.api';
+import type { NetworkId } from '$lib/types/network';
+import { mapNetworkIdToBitcoinNetwork, mapToSignerBitcoinNetwork } from '$lib/utils/network.utils';
+import type { Identity } from '@dfinity/agent';
+
+export const loadBtcPendingSentTransactions = async ({
+	address,
+	identity,
+	networkId
+}: {
+	address: string;
+	identity: Identity;
+	networkId: NetworkId;
+}) => {
+	try {
+		const network = mapNetworkIdToBitcoinNetwork(networkId);
+		if (!network) {
+			throw new Error(`Invalid networkId: ${networkId.toString}`);
```

Comment:
- should this service toast an error?

- if you use the `ResultSuccess` pattern I guess this can become a return?

Reply:
Not toast an error (at least for now). The component will show a different error because the store is populated with the error.

Reply:
Regarding returning the error vs throwing it?

Reply:
I was assuming it was ok that this was thrown, got to the catch and return the error.

But I guess I can return here as well.

Reply:
It's ok but, if you also `return` instead of `throw`, then you spare the need of a `try catch` in your consumer component - i.e. you will be able to interpret solely the result.